### PR TITLE
Adjust RecordArray to direct storage.

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Switch from indirect access to direct access. More convenient to use views.
@markdewing I kept the RecordArray as a specialized type of Matrix.

## What type(s) of changes does this code introduce?
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by ye-luo** [comment]
Test this please

File: src/Containers/MinimalContainers/RecordArray.hpp [hunk 1]
```diff
@@ -2,91 +2,54 @@
...
-    return storage_[param_idx * param_dim_ + entry_idx * entry_dim_];
-  }
+  using Base = Matrix<T>;
 
-  RecordArray() : nparam_(0), nentry_(0) {}
+  RecordArray() {}
 
   /// Constructor specifying the number of parameters and entries
 
-  RecordArray(int nparam, int nentry) : nparam_(nparam), nentry_(nentry)
```
Comment (ye-luo) [inline src/Containers/MinimalContainers/RecordArray.hpp:55]: I have not swapped nparam and nentry order. I will push a commit to change it.
The documentation says entries are rows and parameters are columns. So the existing order is error-prone.

Comment (ye-luo) [reply src/Containers/MinimalContainers/RecordArray.hpp:55]: changed.

**Comment by ye-luo** [comment]
Test this please

**Review by prckent — APPROVED** [review]
LGTM

**Pull request merged**